### DISCUSSION TITLE
Define min/max constraint for date type field?

### DISCUSSION THREAD

**Discussion by BenPortner** [frictionlessdata/frictionlessdata.io#583] [Q&A] [answered]
Dear frictionlessdata community,

I am currently realizing my first frictionless project and I have a question:
My resource is a table and one of the columns contains `date`-type data. I would like to specify a constraint on that column, however when I try to set `minimum`or `maximum`, validation fails. What is the correct way to specifiy a `minimum`/`maximum` constraint on `date`-type columns?

My field description looks like this:
```javascript
{
    'name': 'Date',
    'description': 'Reference date between 1950 and 2100.',
    'type': 'date',
    'constraints': {
        'required': True,
        // 'minimum': '1950',
        // 'maximum': '2100',
    }
}
```

Thank you for your help!
Ben

**Comment by lwinfree** [comment]
Hi @BenPortner thanks for this question. I'm tagging @roll who can help answer it. :-)

**Comment by roll** [comment, marked as answer]
Hi @BenPortner,

The constraint should follow the data type and format so for your date column it will be `2020-01-01`. 

BTW there is the `year` data type available - https://specs.frictionlessdata.io/table-schema/#types-and-formats. So you can have:

```
{
    'name': 'Date',
    'description': 'Reference date between 1950 and 2100.',
    'type': 'year',
    'constraints': {
        'required': True,
        'minimum': 1950,
        'maximum': 2100,
    }
}
```
